### PR TITLE
Add some defensive codes of new protobuffers' semantics

### LINKED ISSUE #6736: K8s-mesh analyzer failed to filter invalid log entry in ALS
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [x] Bug
- [ ] Requirement
- [ ] Feature or performance improvement

___
### Bug
Once an invlaid log entry is passed to ALS receiver, The k8s-mesh analyzer failed to filter out this entry.

Oap console log:
```
 DEBUG [] - Unknown ip , ip -> service is null
```
The invalid log entry:
```
log_entry { common_properties { downstream_remote_address { socket_address { address: "10.132.0.22" port_value: 24870 } } downstream_local_address { socket_address { address: "x.x.x.x" port_value: 8443 } } tls_properties { tls_version: TLSv1_2 tls_cipher_suite { value: 52392 } tls_sni_hostname: "httpbin.tetrate.io" local_certificate_properties { subject: "xxxxxxx" } peer_certificate_properties { } tls_session_id: "594f75aaa6e3a3536db341ff96fd41c5a29d4d5416cf7fd7f190cc1bddc936ec" } start_time { seconds: 1618224071 nanos: 368039000 } time_to_last_rx_byte { nanos: 74180 } time_to_first_downstream_tx_byte { nanos: 316918 } time_to_last_downstream_tx_byte { nanos: 334043 } route_name: "httpbin-9381f6dda881ba94-httpbin-tetrate-io-1" downstream_direct_remote_address { socket_address { address: "10.132.0.22" port_value: 24870 } } } protocol_version: HTTP11 request { request_method: GET authority: "xxxx" path: "/success" user_agent: "curl/7.59.0" forwarded_for: "10.132.0.22" request_id: "168e956d-543a-4b56-8a7f-b3971c948a91" request_headers_bytes: 1992 } response { response_code { value: 301 } response_headers_bytes: 105 response_code_details: "direct_response" } }
```

A possible reason is related to  https://github.com/apache/skywalking/blob/64ff0305f7b09fab29771cd2356d477bfb91852f/oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/envoy/als/k8s/K8sALSServiceMeshHTTPAnalysis.java#L142

*upstreamRemoteAddress is not null, but the  socketAddress in it is an empty string.*

### PR DESCRIPTION
**Pull request by kezhenxu94**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #6736 .
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

File: oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/envoy/als/LogEntry2MetricsAdapter.java [hunk 1]
```diff
@@ -97,9 +94,8 @@
     protected ServiceMeshMetric.Builder adaptCommonPart() {
         final AccessLogCommon properties = entry.getCommonProperties();
         final String endpoint = endpoint();
-        final int responseCode = ofNullable(entry.getResponse()).map(HTTPResponseProperties::getResponseCode)
-                                                                .map(UInt32Value::getValue)
-                                                                .orElse(200);
+        int responseCode = entry.getResponse().getResponseCode().getValue();
```
Comment (hanahmily) [inline oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/envoy/als/LogEntry2MetricsAdapter.java:97]: Would not we get null from entry.getResponse?

Comment (kezhenxu94) [reply oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/envoy/als/LogEntry2MetricsAdapter.java:97]: > Would not we get null from entry.getResponse?

It always constructs a new Response object if it’s null

**Review by wu-sheng — APPROVED** [review]
Seems good.

**Review by hanahmily — APPROVED** [review]
LGTM, a nice patch to prevent some potential bugs around proto

**Pull request merged**